cloud/cloud.go

```diff
@@ -757,7 +758,7 @@ func (p *Provider) CheckServer(serverID string) (working bool, err error) {
 		if _, present := p.resources.Servers[serverID]; present {
 			delete(p.resources.Servers, serverID)
 			p.Unlock()
-			err = p.saveResources()
+			err = p.saveResources(p.cloudContext(ctx))
```

Comment:
Some saveResources() calls you call cloudContext(), some you don't. I think none of them need it because saveResources() itself calls cloudContext(). Double-check and make consistent for all calls.